undertow/src/main/java/org/wildfly/extension/undertow/filters/AffinityResourceDefinition.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.undertow.filters;
+
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.ReloadRequiredAddStepHandler;
+import org.jboss.as.controller.ReloadRequiredRemoveStepHandler;
+import org.jboss.as.controller.SimpleResourceDefinition;
+import org.wildfly.extension.undertow.Constants;
+import org.wildfly.extension.undertow.UndertowExtension;
+
+/**
+ * Base class for affinity resources.
+ *
+ * @author Radoslav Husar
+ */
+public abstract class AffinityResourceDefinition extends SimpleResourceDefinition {
+
+    protected static PathElement pathElement(String value) {
+        return PathElement.pathElement(Constants.AFFINITY, value);
+    }
+
+    AffinityResourceDefinition(PathElement path) {
+        super(path, UndertowExtension.getResolver(Constants.HANDLER, Constants.AFFINITY, path.getValue()), new ReloadRequiredAddStepHandler(), new ReloadRequiredRemoveStepHandler());
```

Comment:
Unless I'm mistaken, you don't need to reload the whole server, just restart the services of the parent resource.
You can leverage RestartParentResourceRegistration for this.

Reply:
Nevermind - while simply restarting the parent service is preferable to reloading the entire server, this isn't easily done without significant refactoring, since ModClusterService doesn't properly encapsulate service installation.